packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,302 @@
+   * If you specify PriceClass_All, CloudFront responds to requests for your objects from all CloudFront edge locations.
+   * If you specify a price class other than PriceClass_All, CloudFront serves your objects from the CloudFront edge location
+   * that has the lowest latency among the edge locations in your price class.
+   *
+   * @default PRICE_CLASS_ALL
+   */
+  readonly priceClass?: PriceClass;
+
+  /**
+   * How CloudFront should handle requests that are not successful (e.g., PageNotFound).
+   *
+   *
+   * @default - No custom error configuration.
+   */
+  readonly errorConfigurations?: CfnDistribution.CustomErrorResponseProperty[];
+}
+
+/**
+ * A CloudFront distribution with associated origin(s) and caching behavior(s).
+ *
+ * @experimental
+ */
+export class Distribution extends Resource implements IDistribution {
+
+  /**
+   * Creates a Distribution construct that represents an external (imported) distribution.
+   */
+  public static fromDistributionAttributes(scope: Construct, id: string, attrs: DistributionAttributes): IDistribution {
+    return new class extends Resource implements IDistribution {
+      public readonly domainName: string;
+      public readonly distributionId: string;
+
+      constructor() {
+        super(scope, id);
+        this.domainName = attrs.domainName;
+        this.distributionId = attrs.distributionId;
+      }
+    }();
+  }
+
+  /**
+   * Creates a Distribution for an S3 Bucket, where the bucket has not been configured for website hosting.
+   *
+   * This creates a single-origin distribution with a single default behavior.
+   */
+  public static forBucket(scope: Construct, id: string, bucket: s3.IBucket): Distribution {
+    return new Distribution(scope, id, {
+      defaultBehavior: { origin: Origin.fromBucket(scope, 'SingleOriginBucket', bucket) },
+    });
+  }
+
+  /**
+   * Creates a Distribution for an S3 bucket, where the bucket has been configured for website hosting.
+   *
+   * This creates a single-origin distribution with a single default behavior.
+   */
+  public static forWebsiteBucket(scope: Construct, id: string, bucket: s3.IBucket): Distribution {
+    return new Distribution(scope, id, {
+      defaultBehavior: { origin: Origin.fromWebsiteBucket(scope, 'SingleOriginWebsiteBucket', bucket) },
+    });
+  }
+
+  public readonly domainName: string;
+  public readonly distributionId: string;
+
+  private readonly defaultBehavior: CacheBehavior;
+  private readonly additionalBehaviors: CacheBehavior[] = [];
+
+  private readonly errorConfigurations: CfnDistribution.CustomErrorResponseProperty[];
+  private readonly certificate?: acm.ICertificate;
+
+  constructor(scope: Construct, id: string, props: DistributionProps) {
+    super(scope, id);
+
+    if (props.certificate) {
+      const certificateRegion = Stack.of(this).parseArn(props.certificate.certificateArn).region;
+      if (!Token.isUnresolved(certificateRegion) && certificateRegion !== 'us-east-1') {
```

Comment:
Is this the right way to do this? Also, I was ensure of the right pattern for the other case: this stack as a whole is being deployed to another region and the certificate is being created as part of the stack.

Reply:
I believe so. But i'd like @skinny85 expertise here